source/common/http/conn_manager_impl.cc

```diff
@@ -941,6 +944,7 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilte
 #endif
 
   // Now actually encode via the codec.
+  request_info_.firstDownstreamTxByteSent(std::chrono::steady_clock::now());
```

Comment:
Can we document at the API level that our times generated when bytes are serialized to the appropriate codec?  This is almost certainly a good enough approximation for 99.99% of Envoy users and as long as we've docced up "this won't let you know when your control frames are backed up due to flow control" I think we're good.

Reply:
Do you mean in data-plane-api? I can create a PR there to mention the retry behavior, and I think the [flow control](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/config/filter/accesslog/v2/accesslog.proto#L84) comments already exist.

Reply:
Request redacted - I'd missed the existing comments and as you say they're pretty good :-)